WireAPI/Tests/WireAPITests/APIs/SelfUserAPI/SelfUserAPITests.swift

```diff
@@ -0,0 +1,232 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import SnapshotTesting
+import XCTest
+
+@testable import WireAPI
+
+final class SelfUserAPITests: XCTestCase {
+
+    private var apiSnapshotHelper: APISnapshotHelper<SelfUserAPI>!
+
+    // MARK: - Setup
+
+    override func setUp() {
+        super.setUp()
+        apiSnapshotHelper = APISnapshotHelper { httpClient, apiVersion in
+            let builder = SelfUserAPIBuilder(httpClient: httpClient)
+            return builder.makeAPI(for: apiVersion)
+        }
+    }
+
+    override func tearDown() {
+        apiSnapshotHelper = nil
+        super.tearDown()
+    }
+
+    // MARK: - Request generation
+
+    func testGetSelfUserRequest() async throws {
+        try await apiSnapshotHelper.verifyRequestForAllAPIVersions { sut in
+            _ = try await sut.getSelfUser()
+        }
+    }
+    
+    func testPushSupportedProtocolsRequest() async throws {
+        try await apiSnapshotHelper.verifyRequest(for: [.v5, .v6]) { sut in
+            _ = try await sut.pushSupportedProtocols([.mls])
+        }
+    }
+    
+    // MARK: - Request unsupported endpoint
+    
+    func testPushSupportedProtocols_UnsupportedVersionError_V0() async throws {
```

Comment:
What do you think about testing this also for v1, v2, and v3?

Reply:
done